presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -272,12 +278,58 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
                                 .map(JdbcColumnHandle.class::cast)
                                 .collect(toImmutableList()))
                         .collect(toImmutableList())),
-                OptionalLong.empty(), // limit
+                handle.getLimit(), // limit
+                handle.getSortOrder(), //orderby
                 Optional.of(newColumns.build()));
 
         return Optional.of(new AggregationApplicationResult<>(handle, projections.build(), resultAssignments.build(), ImmutableMap.of()));
     }
 
+    @Override
+    public Optional<TopNApplicationResult<ConnectorTableHandle>> applyTopN(
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            long topNCount,
+            List<SortItem> sortItems,
+            Map<String, ColumnHandle> assignments)
+    {
+        if (!isAllowTopNPushdown(session) || !jdbcClient.supportsTopNPushDown() || !jdbcClient.supportsLimit()) {
+            return Optional.empty();
+        }
+
+        JdbcTableHandle handle = (JdbcTableHandle) table;
+
+        if (!sortItems.stream().allMatch(sortItem -> assignments.containsKey(sortItem.getName()))) {
+            return Optional.empty();
```

Comment:
This should never happen. `sortItems` has to reference columns defined in the assignments. It would be a bug in the engine if this wasn't the case. If anything, we may want to add a `checkArgument` for this case.